Templates/DefaultProject/Template/cmake/EngineFinder.cmake

```diff
@@ -13,66 +13,70 @@
     if(NOT json_error)
-        set(O3DE_ENGINE_NAME_TO_USE ${user_project_engine_name})
+        # Add cmake configure dependency on user/project.json in case the engine overrides change 
+        set_property(DIRECTORY APPEND PROPERTY CMAKE_CONFIGURE_DEPENDS "${O3DE_USER_PROJECT_JSON_PATH}")
+        if(user_project_engine)
+            # Update the project json data so it is available in the O3DE_PROJECT_JSON global property
+            string(JSON o3de_project_json SET "${o3de_project_json}" engine "\"${user_project_engine}\"" )
+            set(project_engine ${O3DE_ENGINE_NAME_TO_USE})
+            set(O3DE_ENGINE_NAME_TO_USE ${user_project_engine})
+        endif()
     endif()
-    string(JSON user_engine_path ERROR_VARIABLE json_error GET ${user_project_json} "engine_path")
 endif()
 
-# Extract optional version specifier in the engine name 
-o3de_get_version_specifier_parts("${O3DE_ENGINE_NAME_TO_USE}" O3DE_ENGINE_NAME_TO_USE O3DE_ENGINE_SPECIFIER_OP O3DE_ENGINE_SPECIFIER_VERSION)
+# Save the project json data with overrides in a global property to avoid performance hit
+# from loading the file multiple times
+set_property(GLOBAL PROPERTY O3DE_PROJECT_JSON ${o3de_project_json})
 
-# for backward compatibility
+# For backwards compatibility
 set(LY_ENGINE_NAME_TO_USE ${O3DE_ENGINE_NAME_TO_USE})
 
-if(CMAKE_MODULE_PATH)
-    foreach(module_path ${CMAKE_MODULE_PATH})
-        if(EXISTS ${module_path}/Findo3de.cmake)
-            o3de_engine_compatible(${module_path}/../ ${O3DE_ENGINE_NAME_TO_USE} ${O3DE_ENGINE_SPECIFIER_OP} ${O3DE_ENGINE_SPECIFIER_VERSION} is_compatible engine_version ${ENGINE_FINDER_VERBOSE})
-            if(is_compatible)
-                return() # Engine being forced through CMAKE_MODULE_PATH
-            endif()
-        endif()
-    endforeach()
+# Option 1: Use engine manually set in CMAKE_MODULE_PATH, old engines will check engine name only
+find_package(o3de MODULE QUIET)
+if(o3de_FOUND)
+    message(STATUS "Selecting engine from CMAKE_MODULE_PATH '${CMAKE_MODULE_PATH}'.")
+    return()
 endif()
 
-# Read the user/project.json file and look for the engine_path
-if(user_engine_path)
-    if(EXISTS ${user_engine_path}/cmake)
-        # always be verbose on this compatibility check
-        o3de_engine_compatible(${user_engine_path} ${O3DE_ENGINE_NAME_TO_USE} ${O3DE_ENGINE_SPECIFIER_OP} ${O3DE_ENGINE_SPECIFIER_VERSION} is_compatible engine_version TRUE)
-        if(is_compatible)
-            message("-- Selecting engine from 'user/project.json' at '${user_engine_path}'")
-            list(APPEND CMAKE_MODULE_PATH "${user_engine_path}/cmake")
-            return()
+# Option 2: Use the engine from the 'engine_path' field in <project>/user/project.json
+if(user_project_json)
+    string(JSON user_project_engine_path ERROR_VARIABLE json_error GET ${user_project_json} engine_path)
+    if(user_project_engine_path AND NOT json_error)
```

Comment:
Can we add a else block as well that logs if the `<project>/user/project.json` file has a json error in it?

Previous experience has shown that users will hand edit these JSON files and they will have errors in them like missing closing quotes or a additional commas.